library/std/src/io/buffered.rs

```diff
@@ -1208,6 +1253,49 @@ mod tests {
         assert_eq!(reader.fill_buf().ok(), Some(&[2, 3][..]));
     }
 
+    #[test]
+    #[unstable(feature = "seek_convenience", issue = "59359")]
+    fn test_buffered_reader_stream_position() {
+        let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
+        let mut reader = BufReader::with_capacity(2, io::Cursor::new(inner));
+
+        assert_eq!(reader.stream_position().ok(), Some(0));
+        assert_eq!(reader.seek(SeekFrom::Start(3)).ok(), Some(3));
+        assert_eq!(reader.stream_position().ok(), Some(3));
+        // relative seeking within the buffer and reading position should keep the buffer
+        assert_eq!(reader.fill_buf().ok(), Some(&[0, 1][..]));
+        assert!(reader.seek_relative(0).is_ok());
+        assert_eq!(reader.stream_position().ok(), Some(3));
+        assert_eq!(reader.buffer(), &[0, 1][..]);
+        assert!(reader.seek_relative(1).is_ok());
+        assert_eq!(reader.stream_position().ok(), Some(4));
+        assert_eq!(reader.buffer(), &[1][..]);
+        assert!(reader.seek_relative(-1).is_ok());
+        assert_eq!(reader.stream_position().ok(), Some(3));
+        assert_eq!(reader.buffer(), &[0, 1][..]);
+        // relative seeking outside the buffer will discard it
+        assert!(reader.seek_relative(2).is_ok());
+        assert_eq!(reader.stream_position().ok(), Some(5));
+        assert_eq!(reader.buffer(), &[][..]);
+    }
+
+    #[test]
+    #[should_panic]
+    #[unstable(feature = "seek_convenience", issue = "59359")]
+    fn test_buffered_reader_stream_position_panic() {
+        let inner: &[u8] = &[5, 6, 7, 0, 1, 2, 3, 4];
+        let mut reader = BufReader::with_capacity(4, io::Cursor::new(inner));
+
+        // cause internal buffer to be filled but read only partially
+        let mut buffer = [0, 0];
+        assert!(reader.read(&mut buffer).is_ok());
+        // rewinding the internal reader will cause buffer to loose sync
+        let inner = reader.get_mut();
+        assert!(inner.seek(SeekFrom::Start(0)).is_ok());
+        // overflow when subtracting the remaining buffer size from current position
+        let _ = reader.stream_position();
```

Comment:
This test doesn't do what it intends. It will panic if _any_ of the assertions fail, not just the last one. Instead of using `should_panic`, you could use `catch_unwind` on the last assert.

Reply:
I see. Replacing the last line to
```
let result = panic::catch_unwind(panic::AssertUnwindSafe(|| reader.stream_position().ok()));
assert!(result.is_err());
```
seems to allow the `should_panic` attribute to be removed and the test to succeed.